source/opt/folding_rules.cpp

```diff
@@ -2055,6 +2074,8 @@ spvtools::opt::FoldingRules::FoldingRules() {
 
   rules_[SpvOpSelect].push_back(RedundantSelect());
 
+  rules_[SpvOpStore].push_back(StoringUndef());
```

Comment:
What OpCopyMemory, OpCopyMemorySized, OpImageWrite and atomic operations that can write?

Reply:
We cannot remove `OpCopyMemory*` instructions because the source is a pointer, and the values being copied is what is pointed to by the source pointer.  The pointer should not be OpUndef, and we cannot tell if the memory the source pointer points to is undefined.

We should be able to remove OpImageWrite instructions, and atomic instruction.  I'll look into adding those.

Reply:
Can't remove atomic stores.  OpImageWrite's will not be removed because the check for volatile was not very clear.